lmcache/experimental/storage_backend/abstract_backend.py

```diff
@@ -0,0 +1,91 @@
+import abc
+from concurrent.futures import Future
+from typing import Optional
+
+import torch
+
+from lmcache.experimental.memory_management import BufferMemoryObj, MemoryObj
+from lmcache.utils import CacheEngineKey
+
+
+class StorageBackendInterface(metaclass=abc.ABCMeta):
+
+    def __init__(
+        self,
+        dst_device: str = "cuda",
+    ):
+        """
+        Initialize the storage backend. 
+
+        :param dst_device: the device where the blocking retrieved KV is stored,
+            could be either "cpu", "cuda", or "cuda:0", "cuda:1", etc.
+
+        :raise: RuntimeError if the device is not valid
+        """
+        try:
+            torch.device(dst_device)
+        except RuntimeError:
+            raise
+
+        self.dst_device = dst_device
+
+    @abc.abstractmethod
+    def contains(self, key: CacheEngineKey) -> bool:
+        """
+        Check whether key is in the storage backend. 
+        """
+        raise NotImplementedError
+
+    @abc.abstractmethod
+    def insert_key(self, key: CacheEngineKey, size: int):
```

Comment:
When will this function be called and who will call this function? Is this more like a callback after "put_task" is done?

Reply:
It's always called in the callback of `put`. Adding an abstract method here since all storage backends need to implement this function.